### ISSUE TITLE
Improvements to HorizontalSharding workflows

### ISSUE THREAD

**Issue by rafael** [vitessio/vitess#4117]
#### Feature Description

- [ ] Current workflow automatically discovers existent overlapping shards and creates a single workflow to perform all these splits. This is creates problems when trying to split big keyspaces (eg going from 64 shards to 128), because a failure in splitting a single shards blocks the whole split. Also, there are steps that are not recoverable, so a single failure means you lost the for all the other shards. We should allow users to be explicit about source/destination shards they want to split in an HorizontalResharding. In this way, they can create a workflow per split. 
- [ ] We should add more granularity to **enable_approvals**. Right now is all or nothing. Instead, it would be useful if users can specify which steps they want to require enable approvals. 
- [x]  At the moment, `SplitDiff` in workflow UI are run sequentially, instead of all shards in parallel. This slows down the process, we should run `SplitDiff` in parallel for all shards.

**Comment by yangxuanjia** [comment]
@rafael , the PR #4157 is wonderful for resharding,  it's excellently to create a workflow per split.
But, I looks the code, you demand the same vtworker for destination shards,  and the min rdonly len for destination in one shard.  I don't know why.  I think to split one shard, we use one rdonly and one vtworker is good, in online env, we often have one master, one slave and one rdonly in one shard. So validate workflow as follow as mandatory requirements which I think whether it's rationality.
`
if len(vtworkers) != len(destinationShards) {
		return fmt.Errorf("there are %v vtworkers, %v destination shards: the number should be same", len(vtworkers), len(destinationShards))
	}
if minHealthyRdonlyTabletsVal, err := strconv.Atoi(minHealthyRdonlyTablets); err != nil || minHealthyRdonlyTabletsVal < splitRatio {
		return fmt.Errorf("there are not enough rdonly tablets in source shards. You need at least %v, it got: %v", splitRatio, minHealthyRdonlyTablets)
	}
`

**Comment by yangxuanjia** [comment]
In fact, we only need one shard resharding error do not block other shards resharding, so can we reduce the restrict, allow we only have one rdonly and one vtworker in one shard when resharding, and if user want to be more explicit, also can use your origin design.

**Comment by mattlord** [comment]
I'm closing this as no longer relevant.
